config/src/main/java/com/yahoo/vespa/config/JRTConnectionPool.java

```diff
@@ -66,19 +66,31 @@ public synchronized JRTConnection getCurrent() {
     }
 
     @Override
-    public synchronized JRTConnection switchConnection() {
+    public synchronized JRTConnection switchConnection(Connection failingConnection) {
         List<JRTConnection> sources = getSources();
         if (sources.size() <= 1) return currentConnection;
 
-        List<JRTConnection> sourceCandidates = sources.stream()
-                                                    .filter(JRTConnection::isHealthy)
-                                                    .collect(Collectors.toList());
-        JRTConnection newConnection;
+        if ( ! currentConnection.equals(failingConnection)) return currentConnection;
```

Comment:
Override `equals()` in JRTConnection or use `==` ?

Reply:
Yup, thanks, will do